dask/dataframe/groupby.py

```diff
@@ -1449,6 +1453,8 @@ def __init__(
         if observed is not None:
             self.observed["observed"] = observed
 
+        # raises a warning about observed=False with pandas>=2.1.
+        # We want to raise here, and not later down the stack.
         self._meta = self.obj._meta.groupby(
             by_meta, group_keys=group_keys, **self.observed, **self.dropna
         )
```

Comment:
Is there a reason we lean towards having pandas emit the future warning? In https://github.com/dask/dask/pull/10094, it seems like the approach taken is to emit a roughly identical warning through Dask (and I presume suppress this warning when it later pops up in pandas)

Reply:
Yes. Up to [this commit](https://github.com/dask/dask/pull/10095/commits/56ade9f14a0369ca66104a4e5f33a2b24cde22fc) I attempted to do the same - emit our own warning and suppress one from `pandas`, but the code to find out if we're grouping on categoricals quickly turned complex. Since `pandas` is going to do the same work anyway, and we need a warning in the same place, I opted for simpler code and delegated the warning.

Reply:
Thanks for the clarification! That makes sense to me - I wonder if there's a way to catch this warning and emit on the Dask end, but that's only if we want to be super intentional about where the warnings are coming from, and I'd imagine wouldn't be worthwhile otherwise.

Reply:
I thought about that, it's doable, but I don't think it would serve much purpose. Perhaps if we needed to reword it, but the wording is currently fine.